go/enclave/core/batchlimiter.go

```diff
@@ -0,0 +1,107 @@
+package core
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// BatchSizeLimiter - Acts as a limiter for batches based
+// the data from the transaction that we have to publish to the l1.
+// Acts as a calldata reservation system that accounts for both
+// transactions and cross chain messages.
+type batchSizeLimiter struct {
+	Size            uint64
+	ContractAddress common.Address
+	Topic           common.Hash
+}
+
+type BatchSizeLimiter interface {
+	AcceptTransaction(tx *types.Transaction) error
+	ProcessReceipt(receipt *types.Receipt) error
+}
+
+var ErrInsufficientSpace = errors.New("insufficient space in BatchSizeLimiter")
+
+// BatchMaxTransactionData - The number where we will cut off processing transactions inside the evm facade.
+const BatchMaxTransactionData = 25_000
```

Comment:
Are these bytes ?

Reply:
yep

Reply:
Optional, but would rename to BatchMaxTransactionDataSize